AppDelegate.swift

```diff
@@ -1,141 +1,378 @@
+                let managedContext = self.persistentContainer.viewContext
+                if let oldFetchRequest = Encounter.fetchOldEncounters() {
+                    let batchDeleteRequest = NSBatchDeleteRequest(fetchRequest: oldFetchRequest)
+                    do {
+                        try managedContext.execute(batchDeleteRequest)
+                        // set cleaned date on success only
+                        UserDefaults.standard.setValue(today, forKey: "lastTimeDataCleaned")
+                    } catch {
+                        // old data deletion failed!
+                        // since the lastTimeDataCleaned is not set it will be retried hopefully same day and succeed
+                        
+                        //fatalError("Failed to execute request: \(error)")
+                    }
+                    self.endBackgroundTask()
+                }
+            }
+        }
+    }
+    
+    let blackScreenTag = 123
+    
+    private func showBlackscreen() {
+        if window?.viewWithTag(blackScreenTag) == nil {
+            let powerSavingView = UIView()
+            powerSavingView.frame = window!.frame
+            powerSavingView.tag = blackScreenTag
+            powerSavingView.contentMode = .scaleAspectFit
+            powerSavingView.backgroundColor = .black
+            powerSavingView.alpha = 0
+
+            let appNameImage = UIImageView(image: UIImage(named: "lowPowerLogo"))
+            appNameImage.frame = powerSavingView.frame
+            appNameImage.contentMode = .center
+            powerSavingView.addSubview(appNameImage)
+
+            window?.addSubview(powerSavingView)
+            UIView.animate(withDuration: 0.5) {
+                powerSavingView.alpha = 1
+            }
+            
+            NotificationCenter.default.post(name: .disableUserInteraction, object: nil)
+            NotificationCenter.default.post(name: .enableDeferringSystemGestures, object: nil)
+        }
+    }
+    
+    private func dismissBlackscreen() {
+        if window?.viewWithTag(blackScreenTag) != nil {
+            let powerSavingView = window?.viewWithTag(blackScreenTag)
+            powerSavingView?.alpha = 0
+            powerSavingView?.removeFromSuperview()
+            
+            NotificationCenter.default.post(name: .enableUserInteraction, object: nil)
+
+            NotificationCenter.default.post(name: .disableDeferringSystemGestures, object: nil)
+        }
+    }
+    
+    var sampleAngleY = [Double]()
+    var sampleAngleZ = [Double]()
+    
+    fileprivate func appendYSample(sample: Double)  {
+        sampleAngleY.append(sample)
+        if(sampleAngleY.count > 10){
+            sampleAngleY.removeFirst()
+        }
+    }
+    
+    fileprivate func appendZSample(sample: Double)  {
+        sampleAngleZ.append(sample)
+        if(sampleAngleZ.count > 10){
+            sampleAngleZ.removeFirst()
+        }
+    }
+    
+    func startAccelerometerUpdates() {
```

Comment:
This appears to be where they implement their "low power" mode when face down on the desk. How much power are they using when not face down on the desk getting constant accelerometer updates?